src/Microsoft.Diagnostics.Monitoring.EventPipe/Counters/TraceEventExtensions.cs

```diff
@@ -329,14 +336,55 @@ private static void HandleMultipleSessionsNotSupportedError(TraceEvent obj, stri
             }
         }
 
-        private static void HandleObservableInstrumentCallbackError(TraceEvent obj, string sessionId, out ICounterPayload payload)
+        private static void HandleMultipleSessionsConfiguredIncorrectlyError(TraceEvent obj, string sessionId, out ICounterPayload payload)
         {
             payload = null;
 
             string payloadSessionId = (string)obj.PayloadValue(0);
-            string error = (string)obj.PayloadValue(1);
 
             if (payloadSessionId != sessionId)
+            {
+                // If our session is not the one that is running then the error is not for us,
+                // it is for some other session that came later
+                return;
+            }
+
+            string expectedMaxHistograms = (string)obj.PayloadValue(1);
+            string actualMaxHistograms = (string)obj.PayloadValue(2);
+            string expectedMaxTimeSeries = (string)obj.PayloadValue(3);
+            string actualMaxTimeSeries = (string)obj.PayloadValue(4);
+            string expectedRefreshInterval = (string)obj.PayloadValue(5);
+            string actualRefreshInterval = (string)obj.PayloadValue(6);
+
+            StringBuilder errorMessage = new("Error: Another shared metrics collection session is already in progress for the target process, perhaps from another tool? " + Environment.NewLine +
```

Comment:
Instead of providing a generic message, I wanted something that gives specific steps for how to resolve the issue - this will state which values are incorrect (compared to the existing SHARED session), and what values need to be used.

Reply:
Can you remove the "perhaps from another tool?" part? I like the message's intent but that part is too informal for a diagnostic message

Reply:
I actually pulled that bit directly from the `HandleMultipleSessionNotSupportedError` (https://github.com/dotnet/diagnostics/blob/a141a4708e384a76aa06c3d4fd117f40a06c6792/src/Microsoft.Diagnostics.Monitoring.EventPipe/Counters/TraceEventExtensions.cs#L332) - would you want me to change that message as well then, or just leave them as is to be consistent?

Reply:
I would like you to change the existing message to be consistent, thanks.